tools/pub_get_offline.py

```diff
@@ -71,21 +71,24 @@ def CheckPackage(package):
 
 
 def Main():
-  leading = os.path.join(
-      "src", "third_party", "dart", "tools", "sdks", "dart-sdk", "bin"
-  )
+  script = os.path.realpath(__file__)
+  engine_src = os.path.normpath(os.path.join(
+      os.path.dirname(script), "..", ".."))
+  dart_sdk_bin = os.path.join(engine_src,
+      "third_party", "dart", "tools", "sdks", "dart-sdk", "bin")
   dart = "dart"
   if os.name == "nt":
     dart = "dart.exe"
   pubcmd = [
-      os.path.abspath(os.path.join(leading, dart)), "pub", "get", "--offline"
+      os.path.join(dart_sdk_bin, dart), "pub", "get", "--offline"
   ]
 
   pub_count = 0
   for package in ALL_PACKAGES:
-    if FetchPackage(pubcmd, package) != 0:
+    package_path = os.path.join(engine_src, ".." , package)
```

Comment:
I could modify the package paths to remove the need for .., or I could create the array of package paths based on engine_src, moving it to a global variable ENGINE_SRC, if that is better.